src/Ng/Telemetry/JobPropertiesTelemetryInitializer.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using NuGet.Services.Metadata.Catalog;
+using System;
+
+namespace Ng
+{
+    public class JobPropertiesTelemetryInitializer : ITelemetryInitializer
```

Comment:
The idea here is that a job can define at runtime which properties it wants to add to all telemetry.